### PR TITLE
Fixes job list alignment

### PR DESCRIPTION
**Pull request by franz899**
- It fixes the first li inside a grav-c-job-list > ul

### CODE REVIEW

File: src/ui-lib/sass/05-components/03-organisms/_job-list.scss [hunk 1]
```diff
@@ -25,7 +25,7 @@
     }
   }
 
-  li {
+  li:not(:first-child) {
```
Comment (danbull) [inline src/ui-lib/sass/05-components/03-organisms/_job-list.scss]: I think this is fine but I've seen `li + li` (or similar) being used also. Is there a convention @ThePeach @james-nash?

Comment (ThePeach) [reply src/ui-lib/sass/05-components/03-organisms/_job-list.scss]: 👍 second the `li + li`. Btw no convention here.

Comment (franz899) [reply src/ui-lib/sass/05-components/03-organisms/_job-list.scss]: I think that `:not` follows more the css standards, if there is a rule that does exactly what we need why should we use another way, even if it accomplish the same thing.

I have no strong opinion about it.

Comment (ThePeach) [reply src/ui-lib/sass/05-components/03-organisms/_job-list.scss]: They both follow standard, only difference is that `li + li` is a CSS pattern that follows the lobotomised owl pattern and is quite widely used.

Comment (franz899) [reply src/ui-lib/sass/05-components/03-organisms/_job-list.scss]: Those are techniques though, not specific rules.

I know about the owl, and it is useful, but is a bit tricky to respect as a rule. There idea is to not to specify the margin on the single elements, but it has already been broken in this css.

If we want to keep it consistent then I propose this:
```
.grav-c-job-list {
  @media all and (min-width: gravy-breakpoint(regular)) {
    display: grid;
    grid-gap: $grav-sp-m;
    grid-template-columns: 30% 1fr;
  }

  > header > p {
    @include typi(small-text);
    margin-top: 0;
  }

  > ul {
    padding-left: 0;
    list-style-type: none;

    @media all and (min-width: gravy-breakpoint(small)) {
      columns: 2;
    }

    @media all and (min-width: gravy-breakpoint(regular)) {
      margin-top: 0;
    } 
  }

  li + li {
    margin-top: $grav-sp-s;
  }
}

* + .grav-c-job-list {
  margin-top: $grav-sp-xl;
}
```

Comment (ThePeach) [reply src/ui-lib/sass/05-components/03-organisms/_job-list.scss]: sounds good to me, the last one can be written inside the main class as:

```
.grav-c-job-list {
  ...

  * + & {
    margin-top: $grav-sp-xl;
  }
}
```

Comment (franz899) [reply src/ui-lib/sass/05-components/03-organisms/_job-list.scss]: Oh yeah of course, I copied my rule from Firefox.

**Pull request merged**